### ISSUE TITLE
Weapon Type + Burn matrix

### ISSUE THREAD

**Issue by zacgoodwin** [DestinyItemManager/DIM#7790]
### Proposed change

I'd like to see the ability to quickly view of what weapon you have flagged as Keep / Favorite for each type of weapon and element

![image](https://user-images.githubusercontent.com/666712/151075002-e2704a2a-85e7-487e-9ae5-c023c6ad444a.png)

The matrix could be customizable by a couple dimensions:
- Weapon Type
- Weapon archetype 
- Weapon Element
- Selected weapon perks (Demolitionist, Wellspring, Thresh, Etc.)
- Engagement Distance (Short, Medium, Long) This would be subjective but could be handy (nice to have?)
- Engagement Type (Rapid, Aggressive, Precise) to match the charged with light + VOG mods (nice to have?)

Functionally speaking, I view this a bit like a pivot table in excel when you can drag and drop the options into X. Y and Filters to really get a feel visually for what you have in your vault

Clicking on a weapon name could expand out to show you all the rolls you have of that weapon 

![image](https://user-images.githubusercontent.com/666712/151076349-b2f07e51-535c-40f9-92fd-16362cb647b7.png)

### How does this fit into your workflow?

This would useful to be prepared for high level content. It could help you optimize your vault and what rolls to hunt for to cover the following cases:
- Burns on lost sectors rotate frequently so you want to make sure you have all the elements covered
- Updates to Destiny changes the meta to benefit weapons / archtypes so you want to visually see what fits that criteria and what you're missing
- Seasonal artifacts encourage the use of types of weapons, at a glance you can see if you're prepared or not
- Perks lend themselves to builds, for example demolitionist, and so you can quickly pick the right weapon for the build in what ever 

I view this as the weapon companion to the loadout builder and would help visually minded organizers like myself.

**Comment by bhollis** [comment]
Have you tried the Organizer?

**Comment by zacgoodwin** [comment]
I have, the best analogy I can think of is that the organizer is my spreadsheet where as what I am proposing is a pivot table / reporting. I can ask the organizer what I have then drill into that section where things are missing, but it's hard to see at a glance what I am missing across my entire collection. 

![image](https://user-images.githubusercontent.com/666712/151195290-46a442d1-1bcc-4099-a633-625c292d747d.png)

Being able to bucketize things at a glance would be a nice way to replace the inventory screen for certain functions as well. The ability to sort is great but I wish I could display a second dimension on it. I had to abandon using the favorite tag because I couldn't get a sort order together that I liked where the favorites were easy to pick out and I was able to find the weapon/energy type combination I needed. The sort order I have 

![image](https://user-images.githubusercontent.com/666712/151196119-8914f2be-5f09-4a0c-9df1-532fb60ee9a5.png)

What I did to work around this was make a load out with my favorite weapons by activity: 

![image](https://user-images.githubusercontent.com/666712/151196429-f34df567-1ee9-40e3-84f5-1236cbb250d8.png)

Hope this helps clarify.

**Comment by zacgoodwin** [comment]
https://destinyrecipes.com/vault

This is also something along the similar lines, but specifically to be used for cleaning the vault instead of analysis :)

**Comment by bhollis** [comment]
@vivekhnz you maybe wanted to look into this?

**Comment by vivekhnz** [comment]
Yep, I'm interested in working on something like this. I have a similar workflow whereby I triage weapons based on weapon type + slot + archetype for PvP and weapon type + damage type for PvE using the organizer. It's fairly time-consuming and having a way to group items into buckets (like the proposed matrix) would streamline the process.

I am leaning towards incrementally modifying Organizer rather than building a new page/tool to accomplish this. Some areas I think we should look at are:
- **Reduce the rigidity of top-level Organizer filters** - the top-level weapon type filters (e.g. **Auto Rifles**) are great when getting started but force you to do a lot of clicking if you want to triage weapons across many weapon types. It would be nice if clicking **Weapon** would list all weapons and then I can use the search box to filter down to certain weapon types e.g. `is:kineticslot (is:autorifle or is:pulserifle or is:scoutrifle or is:bow)` if I want to see all my long-range primary weapons in the Kinetic slot.
- **Group items by a column** - add an additional dropdown to select a column that items should be grouped by. This would only be valid for certain columns (e.g. Type, Archetype, Damage, Tag, Locked etc.). We would still display items in a list but their would be a 'spacer' at the start of each group. Sorting would be applied to each group rather than against all items that match the filter. This effectively gives us a 'secondary sort' criteria - group by Type, sort by Archetype to see all my weapons in a list, sorted by their archetype without needing to click the filter buttons to move between weapon types.
- **Add a second grouping criteria for matrix view** - add an additional grouping dropdown with the same options as the previous point. When a value is selected for both, the Organizer list view is replaced with a grid whereby each cell represents a bucket. Rows correspond to each value of the first grouping criteria and columns correspond to each value of the second grouping criteria. Each bucket (grid cell) contains the item tiles for all items that match both criteria (as well as the search filter).
- **Multi-select items** - closely related to #7209, we should be able to shift-click item tiles in grid cells to add them to the current selection. We can then perform bulk actions (e.g. tagging, moving etc.) using the existing action buttons in Organizer.

This feels fairly ambitious to me so would likely be a long-term effort. One could also argue that this should be a separate tool outside of DIM or that users can use CSV export and build this functionality in their own spreadsheet tool. I've tried both approaches in the past and have given up because DIM is really good at providing tools to figure out how good an item is (wishlists, perk tooltips, notes, trait and archetype classification).

**Comment by BisonThrow** [comment]
I am also looking for a similar functionality.  I want to be able to populate a chart similar to what i have attached.
![2022-06-15_12-12-14](https://user-images.githubusercontent.com/107570931/173876777-4449a87f-4375-490c-a957-e62092d69ebb.jpg)

This way i can easily see where i am missing element/archetype coverage

**Comment by bhollis** [comment]
> Reduce the rigidity of top-level Organizer filters

It might also be nice if you could shift-click to select multiple weapon types? The main reason this doesn't work today is that not all weapon types are meaningfully comparable.

> Group items by a column

Not sure I follow - you can already sort by N different columns by shift-clicking the headers. Having dividers or backgrounds between the primary sort might be cool, but you'd maybe want to tie it to some cardinality analysis so you're not just putting dividers between every item.

> Add a second grouping criteria for matrix view

Sounds like a pivot table? I'm onboard but I'd need to see what it looks like

> Multi-select items

You can already do this with the checkboxes - what am I missing?

**Comment by vivekhnz** [comment]
> shift-click to select multiple weapon types

Yep, I like that.

> sort by N different columns by shift-clicking the headers

Oh wow, didn't realise that was already a feature!

> pivot table?

Yeah, it's basically a pivot table although the idea is each cell shows all matching items instead of a summary (e.g. count, sum etc.) like you would typically use a pivot table for.

> You can already do [multi-select] this with the checkboxes - what am I missing?

Reusing the checkboxes for multiselect would be OK. My thought was that we could reuse any multiselect styling done for #7209 - if it works in a high density layout like the item grid, it should work fine in a matrix grid cell.

Something like this is what I have in mind:
![dim-organizer-matrix-mockup](https://user-images.githubusercontent.com/17512262/174037470-13870be0-ae05-4cc3-9e31-50c517638347.png)

Another option could be to display items using the tiles from #8248. That would provide more 'at-a-glance' info about an item, at the cost of density.

**Comment by vivekhnz** [comment]
Yep, a 'group by' toggle would be OK to start with. Maybe we could look at fancier things like the ability to drag a column header out to a 'grouping area' to switch from list view to grid view. Not totally sure though, happy to keep it simple to start with at least.

As a side-note, the matrix view may be an interesting place to surface 'gaps' in your collection along with how you could fill out those gaps. For example, if you don't have any Stasis Hand Cannons, DIM could optionally show a list of Stasis Hand Cannons from Collections inside that cell instead (and their source in a tooltip) or ones that are currently on sale from a vendor. If we allow grouping on specific perks, it could be used to answer questions like "_Do I have any Solar Hand Cannons with Demolitionist? If not, where can I get one that does?_".

**Comment by DCForever** [comment]
It looks like this issue is old, but this is also something I would really like.
